portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -79,6 +89,17 @@ private String convertToJsonUsingGson(List<Comment> comments) {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // Get the input from the form.
+    String max = getParameter(request, "max-comments", null);
+    // If a maximum number of comments has been selected, only update the maxComments variable and return.
+    if (max != null) {
+
+        maxComments = Integer.parseInt(max);
```

Comment:
What happens if the user passes a non-integer string? What about a negative number?

Reply:
The user technically can only choose for the maximum number of comments to be either 5, 10, 50, or 100. That is in the form of a dropdown select form, where 10 is initially selected by default.

Reply:
Even though there's no input box, a user could still manually send the query through the URL string. In general you can't trust user input. If somebody sent a crafted URL with a non-integer request parameter your server might crash

Reply:
Oh I see! I did not think about that. I just added the necessary input checks in a new commit to this PR.